packages/graphql-server-restify/src/restifyApollo.ts

```diff
@@ -0,0 +1,90 @@
+import * as restify from 'restify';
+import * as graphql from 'graphql';
+import * as url from 'url';
+import { GraphQLOptions, runHttpQuery, HttpQueryError } from 'graphql-server-core';
+import * as GraphiQL from 'graphql-server-module-graphiql';
+
+export interface RestifyGraphQLOptionsFunction {
+  (req?: restify.Request, res?: restify.Response): GraphQLOptions | Promise<GraphQLOptions>;
+}
+
+// Design principles:
+// - You can issue a GET or POST with your query.
+// - simple, fast and secure
+//
+
+export interface RestifyHandler {
+  (req: restify.Request, res: restify.Response, next): void;
+}
+
+export function graphqlRestify(options: GraphQLOptions | RestifyGraphQLOptionsFunction): RestifyHandler {
+  if (!options) {
+    throw new Error('Apollo Server requires options.');
+  }
+
+  if (arguments.length > 1) {
+    // TODO: test this
+    throw new Error(`Apollo Server expects exactly one argument, got ${arguments.length}`);
+  }
+
+  return async (req: restify.Request, res: restify.Response, next) => {
+    runHttpQuery([req, res], {
+      method: req.method,
+      options: options,
+      query: req.method === 'POST' ? req.body : req.query,
+    }).then((gqlResponse) => {
+      res.setHeader('Content-Type', 'application/json');
+      res.write(gqlResponse);
+      res.end();
+    }, (error: HttpQueryError) => {
+      if ( 'HttpQueryError' !== error.name ) {
+        throw error;
+      }
+
+      if ( error.headers ) {
+        Object.keys(error.headers).forEach((header) => {
+          res.setHeader(header, error.headers[header]);
+        });
+      }
+
+      res.statusCode = error.statusCode;
+      res.write(error.message);
+      res.end();
+    });
+  };
+}
+
+function isOptionsFunction(arg: GraphQLOptions | RestifyGraphQLOptionsFunction): arg is RestifyGraphQLOptionsFunction {
```

Comment:
I think this function here can be removed
It moved to core.. please check that ;)